### PR TITLE
Refactor prometheus secret handling for multi cluster

### PR DESCRIPTION
**Pull request by lake-of-dreams**
# Description

Currently to register a cluster as managed cluster in admin cluster, a prometheus secret is created which contains the prometheus host for the managed cluster and also the CA Certificate for the managed cluster. This CA certificate is also required for proxying requests from admin cluster to managed cluster ans was used in https://github.com/verrazzano/verrazzano/pull/1189. Since prometheus host of managed cluster and CA Certificate of managed cluster are separate entities and should be made available separately in the admin cluster instead of combining them in a secret with a customised format. This change is to specify CA secret as part of the spec of the vmc resource and prometheus host is updated by Multicluster agent in status field of vmc resource.

This change is dependent on https://github.com/verrazzano/verrazzano/pull/1239 to be merged first for multi cluster api to work.

Fixes VZ-2715

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Comment by mgianatagh** [comment]
Once these changes are merged to master, these instructions will need to be updated:

https://verrazzano.io/docs/setup/multicluster/multicluster/

File: platform-operator/controllers/clusters/vmc_controller_test.go [hunk 1]
```diff
@@ -1591,9 +1564,21 @@ func getScrapeConfig(prometheusYaml string, name string) (*gabs.Container, error
...
+
+// getPrometheusHost returns the prometheus host for testManagedCluster
+func getPrometheusHost() string {
+	return "prometheus.vmi.system.default.1.2.3.4.nip.io"
+}
+
+// getPrometheusHost returns the prometheus host for testManagedCluster
+func getCaCrt() string {
+	return "    -----BEGIN CERTIFICATE-----\n" +
+		"    MIIBiDCCAS6gAwIBAgIBADAKBggqhkjOPQQDAjA7MRwwGgYDVQQKExNkeW5hbWlj\n" +
```
Comment (markxnelson) [inline platform-operator/controllers/clusters/vmc_controller_test.go:1580]: what is this data when decoded?

Comment (lake-of-dreams) [reply platform-operator/controllers/clusters/vmc_controller_test.go:1580]: this string already existed in test, decodes to some jumble characters
```bash
00.0
*H=0;10U
dynamic
```

Comment (lake-of-dreams) [reply platform-operator/controllers/clusters/vmc_controller_test.go:1580]: my bad, I base64 decode it, the cert is not decodable at all..

**Review by markxnelson — APPROVED** [review]

**Pull request merged**